internal/generate/clusterserviceversion/clusterserviceversion.go

```diff
@@ -119,7 +146,33 @@ func (g *Generator) Generate(opts ...Option) (err error) {
 
 	// Add extra annotations to csv
 	g.setAnnotations(csv)
-
+	// If a reader is set, and there is a flag to not update createdAt, then
+	// set the CSV's createdAt to the previous CSV's createdAt if its the only change.
+	if g.ignoreIfOnlyCreatedAtChanged && g.getReader != nil {
```

Comment:
This shouldn't happen of `overwrite` is set to false. If overwrite is explicitly set to false, and `ignoreIfOnlyCreatedAtChanged` is passed then we should possibly error out. If not, that defeats the purpose of us tracking `createdAt` to see the latest changes in bundles. It would be helpful to add that check here.

Reply:
+1